### PR TITLE
Created use case for registration new company

### PR DESCRIPTION
**Pull request by 04tudor**

### CODE REVIEW

File: application/src/main/java/md/maib/retail/RegisterCampaign.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    private boolean validateRange() {
+        if (startInclusive == null || endExclusive == null) {
+            return true;
+        }
+        return startInclusive.isBefore(endExclusive);
+    }
+
+    private static final Validator<RegisterCampaign> validator = ValidatorBuilder
+            .<RegisterCampaign>of()
+            ._object(RegisterCampaign::metaInfo, "metaInfo", c -> c.notNull().message(BLANK))
```
Comment (DemiusAcademius) [inline application/src/main/java/md/maib/retail/RegisterCampaign.java]: message must be NOTNULL

File: application/src/main/java/md/maib/retail/application/register_newcampaign/RegisterCampaignService.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                })
+                .toList();
+
+        var campaign = new Campaign(id, metaInfo, interval,state,loyaltyEventType, rules);
+
+        if (campaigns.add(campaign)) {
+           
+            return right(id);
+        }
+        return left(new UseCaseProblemConflict("CampaignWithSameNameAlreadyExists"));
```
Comment (DemiusAcademius) [inline application/src/main/java/md/maib/retail/application/register_newcampaign/RegisterCampaignService.java]: We don't have campaign name, so we can't have conflicts,.
Lets simply return company id

File: application/src/main/java/md/maib/retail/application/register_newcampaign/RegisterCampaignValidate.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+import java.time.LocalDate;
+import java.util.List;
+
+import static io.vavr.control.Either.left;
+import static io.vavr.control.Either.right;
+
+@Getter
+@RequiredArgsConstructor  // Removed AccessLevel.PRIVATE
+@ToString
+public final class RegisterCampaignValidate {
```
Comment (DemiusAcademius) [inline application/src/main/java/md/maib/retail/application/register_newcampaign/RegisterCampaignValidate.java]: We already have RegisterCampaign command. Move it to register_newcampaign package and use it

File: domain-model/src/main/java/md/maib/retail/model/campaign/Campaign.java [hunk 4]
```diff
@@ -26,12 +28,13 @@ public final class Campaign {
...
 
-            public boolean activate() throws Exception {
+
+            public boolean activate() throws CampaignActivationException {
                 if (state == CampaignState.DRAFT) {
                     this.state = CampaignState.ACTIVE;
                     return true;
                 } else {
-                    return false;
+                    throw new CampaignActivationException("Campaign cannot be activated because it is not in DRAFT state.");
```
Comment (DemiusAcademius) [inline domain-model/src/main/java/md/maib/retail/model/campaign/Campaign.java]: we use exceptions only for exceptional cases. E.g. can not connected to database or other IO exception
for business cases use boolean or Optional or Either

Comment (04tudor) [reply domain-model/src/main/java/md/maib/retail/model/campaign/Campaign.java]: i have to delete exception and to make return as a false?

**Pull request merged**